pkg/controller/masteruserrecord/sync.go

```diff
@@ -185,7 +186,7 @@ func (s *Synchronizer) alignReadiness() (bool, error) {
 	if condition.IsNotTrue(s.record.Status.Conditions, toolchainv1alpha1.MasterUserRecordUserProvisionedNotificationCreated) {
 		notification := &toolchainv1alpha1.Notification{
 			ObjectMeta: v1.ObjectMeta{
-				Name:      s.record.Name + "-provisioned",
+				Name:      s.record.Name + "-provisioned" + "-" + strconv.FormatInt(v1.Now().UnixNano(), 10),
```

Comment:
You can use `GenerateName` instead of `Name` to generate a random resource name. See https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#idempotency

Reply:
Yes, I agree with Alexey, we should rather use the `GenerateName` instead of adding a timestamp suffix

Reply:
updated to use GenerateName here: 1ed72f7ee9951af397509be0b324422c81ce6ad4